test/functional/apps/discover/_discover.js

```diff
@@ -104,25 +106,176 @@ define(function (require) {
           ];
           return common.sleep(4000)
           .then(function () {
-            return common.tryForTime(20 * 1000, function tryingForTime() {
-              return discoverPage.getBarChartData()
-              .then(function compareData(paths) {
-                // the largest bars are over 100 pixels high so this is less than 1% tolerance
-                var barHeightTolerance = 1;
-                for (var y = 0; y < expectedBarChartData.length; y++) {
-                  common.debug(y + ': expected = ' + expectedBarChartData[y] + ', actual = ' + paths[y] +
-                   ', Pass = ' + (Math.abs(expectedBarChartData[y] - paths[y]) < barHeightTolerance));
-                };
-                for (var x = 0; x < expectedBarChartData.length; x++) {
-                  expect(Math.abs(expectedBarChartData[x] - paths[x]) < barHeightTolerance).to.be.ok();
-                }
-              });
-            });
+            return verifyChartData(expectedBarChartData);
+          })
+          .catch(common.handleError(this));
+        });
+
+        bdd.it('should show correct time range string in chart', function () {
+          var expectedTimeRangeString = fromTimeString + ' - ' + toTimeString;
+          return discoverPage.getChartTimespan()
+          .then(function (actualTimeString) {
+            expect(actualTimeString).to.be(expectedTimeRangeString);
+          })
+          .catch(common.handleError(this));
+        });
+
+        bdd.it('should show correct initial chart interval of 3 hours', function () {
+          var expectedChartInterval = 'by 3 hours';
+          return discoverPage.getChartInterval()
+          .then(function (actualInterval) {
+            expect(actualInterval).to.be(expectedChartInterval);
+          })
+          .catch(common.handleError(this));
+        });
+
+        bdd.it('should show correct data for chart interval Hourly', function () {
+          var chartInterval = 'Hourly';
+          var expectedBarChartData = [ '0', '0', '0', '0', '0', '0', '0', '0', '0', '0',
+            '0', '0', '0', '0', '0', '0', '0', '0', '1.2763636363636266', '1.914545454545447',
```

Comment:
No, they could be rounded down to tenths place and be just fine.  I just wouldn't round them down to ints.  They have all that precision because I'm just copying the actual data (from ```console.log(paths)``` as the new expected result.  I just didn't manually round them.  I guess the best thing might be to round to 1 decimal place in the getChartData method so the returned array has the rounded values.  I'd like to do that as a separate PR so that this one doesn't get too big.